op-node/rollup/types.go

```diff
@@ -50,7 +50,29 @@ type Genesis struct {
 	SystemConfig eth.SystemConfig `json:"system_config"`
 }
 
+type ForkName string
+
+const (
+	Regolith ForkName = "regolith"
+	Canyon   ForkName = "canyon"
+	Delta    ForkName = "delta"
+	Ecotone  ForkName = "ecotone"
+	Fjord    ForkName = "fjord"
+	Interop  ForkName = "interop"
+	None     ForkName = "none"
+)
+
+var nextFork = map[ForkName]ForkName{
+	Regolith: Canyon,
+	Canyon:   Delta,
+	Delta:    Ecotone,
+	Ecotone:  Fjord,
+	Fjord:    Interop,
+	Interop:  None,
+}
+
 type Config struct {
+	nextFork ForkName
```

Comment:
Can we move this over to the `chainspec` object? I don't like embedding non-seralized fields here

Reply:
As far as I can tell the engine controller does not currently recognize a `chainspec` object. Are you suggesting that we change the type [here](https://github.com/ethereum-optimism/optimism/blob/develop/op-node/rollup/derive/engine_controller.go#L53) from `*rollup.Config` --> `*rollup.ChainSpec`?

Reply:
I'm suggesting we move this field into `rollup.ChainSpec`. We will need to create an instance of the chain spec from the config inside the engine controller

Reply:
Done in https://github.com/ethereum-optimism/optimism/pull/10530/commits/710830ba1523ecb918e8a5a32551ee6d518f41b1